scripts/asr_language_modeling/ngram_lm/train_kenlm.py

```diff
@@ -54,95 +53,148 @@
     )
+
     parser.add_argument(
-        "--nemo_model_file",
+        "--tokenizer_model_file",
         required=True,
         type=str,
-        help="The path of the '.nemo' file of the ASR model or name of a pretrained model",
+        help="The path to '.model' file of the SentencePiece tokenizer, or '.nemo' file of the ASR model, or name of a pretrained NeMo model",
     )
     parser.add_argument(
         "--kenlm_model_file", required=True, type=str, help="The path to store the KenLM binary model file"
     )
     parser.add_argument("--ngram_length", required=True, type=int, help="The order of N-gram LM")
+    parser.add_argument(
+        "--ngram_prun",
+        default="0",
+        required=False,
+        nargs="+",
+        type=str,
+        help="Whitespace separated digits. Example: 0 0 1. See Pruning section on the https://kheafield.com/code/kenlm/estimation ",
+    )
     parser.add_argument("--kenlm_bin_path", required=True, type=str, help="The path to the bin folder of KenLM")
+    parser.add_argument(
+        "--cache_path", required=False, type=str, default=None, help="Cache path to save tokenized files"
+    )
     parser.add_argument(
         "--do_lowercase", action='store_true', help="Whether to apply lower case conversion on the training text"
     )
+
     parser.add_argument(
         '--preserve_arpa', required=False, action='store_true', help='Whether to preserve the intermediate ARPA file.'
     )
+    parser.add_argument(
+        "--rm_punctuation", action='store_true', help="Whether to remove punctuation marks from text",
+    )
+    parser.add_argument(
+        "--separate_punctuation",
+        action='store_true',
+        help="Whether to separate punctuation with the previouse word by space ",
+    )
+    parser.add_argument("--verbose", type=int, default=0, help="Verbose level from 0. Default is 1 ")
+
     args = parser.parse_args()
 
-    """ TOKENIZER SETUP """
-    logging.info(f"Loading nemo model '{args.nemo_model_file}' ...")
+    args.train_path = kenlm_utils.get_train_list(args.train_path)
+
+    if type(args.ngram_prun) == str:
+        args.ngram_prun = [args.ngram_prun]
+
+    tokenizer, encoding_level, is_aggregate_tokenizer = kenlm_utils.setup_tokenizer(args.tokenizer_model_file)
+    encoded_train_files = []
+
+    if args.cache_path:
+        """ DATASET SETUP """
+
+        for file_num, train_file in enumerate(args.train_path):
+            logging.info(
+                f"Encoding the train file '{train_file}' number {file_num+1} out of {len(args.train_path)} ..."
+            )
+
+            cached_files = glob(os.path.join(args.cache_path, os.path.split(train_file)[1]) + "*")
+            encoded_train_file = os.path.join(args.cache_path, os.path.split(train_file)[1] + f"_{file_num}.tmp.txt")
+            if cached_files:
+                if cached_files[0] != encoded_train_file:
+                    os.rename(cached_files[0], encoded_train_file)
```

Comment:
Use shutil.rename, os.rename has issues with multiple disks and different mount paths if I remember correctly